lib/matplotlib/image.py

```diff
@@ -687,6 +687,21 @@ def get_extent(self):
             else:
                 return (-0.5, numcols-0.5, -0.5, numrows-0.5)
 
+    def get_zdata(self, event):
+        """Get the zdata message for a given event"""
+        xmin, xmax, ymin, ymax = self.get_extent()
+        if self.origin == 'upper':
+            ymin, ymax = ymax, ymin
+        arr = self.get_array()
+        data_extent = mtransforms.Bbox([[ymin, xmin], [ymax, xmax]])
+        array_extent = mtransforms.Bbox([[0, 0], arr.shape[:2]])
+        trans = (mtransforms.BboxTransformFrom(data_extent) +
```

Comment:
FWIW There is a BboxTransform which gives you Bbox -> another Bbox without having to go to a unit Bbox.